src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/DeprecationGroup.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bridge\PhpUnit\DeprecationErrorHandler;
+
+final class DeprecationGroup
+{
+    /**
+     * @var int
+     */
+    private $count = 0;
+
+    /**
+     * @var array<string,DeprecationNotice> keys are messages
+     */
+    private $deprecationNotices = [];
+
+    /**
+     * @param string $message
+     * @param string $class
+     * @param string $method
+     */
+    public function addNoticeFromObject($message, $class, $method)
+    {
+        if (!isset($this->deprecationNotices[$message])) {
+            $this->deprecationNotices[$message] = new DeprecationNotice();
+        }
+
+        $this->deprecationNotice($message)->addObjectOccurence($class, $method);
+        $this->addNotice();
+    }
+
+    /**
+     * @param string $message
+     */
+    public function addNoticeFromProceduralCode($message)
+    {
+        $this->deprecationNotice($message)->addProceduralOccurence();
+        $this->addNotice();
+    }
+
+    public function addNotice()
+    {
+        ++$this->count;
+    }
+
+    /**
+     * @param string $message
+     *
+     * @return DeprecationNotice
+     */
+    private function deprecationNotice($message)
```

Comment:
I would inline `$this->addNotice();` here, rename to `addDeprecationNotice`, make this public and remove redundant extra methods `addNoticeFromObject` and `addNoticeFromProceduralCode`

Reply:
This is probably invalidated by my answer to https://github.com/symfony/symfony/pull/35271/files#r367625497

Reply:
It isn't since my comment was wrong, but `addNotice` needs to be public (it is used directly from the `DeprecationErrorHandler.php` in a case where we don't need to hold the information (the case is notices triggered in tests marked with `@legacy`, which do not need to be reported or even stored in memory).